packages/iris-grid/src/IrisGridTableModel.ts

```diff
@@ -78,21 +80,32 @@ class IrisGridTableModel
     return this.table.applyCustomColumns != null;
   }
 
+  getMemoizedKeyColumnSet = memoize(
+    (inputTableKeys?: readonly string[]) =>
+      new Set(inputTableKeys ?? EMPTY_ARRAY)
+  );
+
+  get keyColumnSet(): Set<ColumnName> {
+    return this.getMemoizedKeyColumnSet(this.inputTable?.keys ?? EMPTY_ARRAY);
+  }
+
   getMemoizedFrontColumns = memoize(
```

Comment:
Input should be `readonly ColumnName[]`

Reply:
Shouldn't this be a set though, because the getMemoizedKeyColumnSet returns a Set,  not an array?